test.py

```diff
@@ -0,0 +1,311 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import unittest
+from inspect import cleandoc
+
+from expr_checker import *
+
+
+class Dangerous:
+    pass
+
+
+class Good:
+    a = "hi"
+    evil = Dangerous()
+    eevil = 4
+
+    _secret_stuff = "1+1=3"
+
+    def tell_me_hi(self):
+        return self.a
+
+    def gift_of_satan(self):
+        return 666
+
+    def gather_secret(self):
+        return self._secret_stuff
+
+    def __enter__(self):
+        return Good()
+
+    def __exit__(self, exec_type, exec_value, traceback):
+        pass
+
+
+class ReadOnlyObject:
+    def __init__(self, x):
+        self.set_x_value(x)
+
+    def set_x_value(self, x):
+        self.x = x
+
+
+def check_type(method, value):
+    safe_type = (str, int, type(None), type(range(0)), Good, type(lambda x: x), ReadOnlyObject)
+
+    if type(value) not in safe_type and not (hasattr(value, "__self__") and type(value.__self__) in safe_type):
+        raise ValueError(f"safe_eval didn't like {value}")
+    else:
+        return value
```

Comment:
```suggestion
def check_type(method, value):
    safe_type = (Good, ReadOnlyObject)

    if type(value) in safe_type:
        return value

    return ast_default_check_type(method, value)
```

Reply:
It won't work if I do this
https://github.com/keyboard-slayer/poc-joda-odoo/pull/1#discussion_r834109887

Reply:
ok, you can forget the `__` :)